libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -127,6 +119,18 @@ string IRGenerator::generate(Block const& _block)
 	return generator.code();
 }
 
+void IRGenerator::collectFunctions(ContractDefinition const& _contract)
+{
+	for (auto const* contract: _contract.annotation().linearizedBaseContracts)
+		for (auto const* fun: contract->definedFunctions())
+			if (fun->isVisibleViaContractTypeAccess())
```

Comment:
One important case that this skips is an internal constructor. I'm not sure constructors should be included here or not. For now I'm assuming that the external ones should and that internal ones should instead be added to the queue when they're called, just like ordinary internal calls. 

I suspect that internal constructor calls are not handled by `endVisit(FunctionCall)` like ordinary internal calls but it wasn't easy to check because `solc --ir` generates no code when the contract has an internal constructor. Is code generation for internal constructors not implemented yet?

So internal constructors are probably not handled properly but since there's no code generated, there's really no difference from the previous behavior.